CoreScriptsRoot/ServerCoreScripts/ClientChat/ChatBar.lua

```diff
@@ -0,0 +1,246 @@
+	local endSize = UDim2.new(1, 0, 1, self.TargetYSize)
+	local curSize = self.GuiObject.Size
+
+	local curAbsoluteSizeY = self.GuiObject.AbsoluteSize.Y
+	self.GuiObject.Size = endSize
+	local endAbsoluteSizeY = self.GuiObject.AbsoluteSize.Y
+	self.GuiObject.Size = curSize
+
+	local pixelDistance = math.abs(endAbsoluteSizeY - curAbsoluteSizeY)
+	local tweeningTime = math.min(1, (pixelDistance * (1 / self.TweenPixelsPerSecond))) -- pixelDistance * (seconds per pixels)
+
+	self.GuiObject:TweenSize(endSize, Enum.EasingDirection.Out, Enum.EasingStyle.Quad, tweeningTime, true)
+end
+
+function methods:FadeOutBackground(duration)
+	self.BackgroundTweener:Tween(duration, 1)
+	--self:FadeOutText(duration)
+end
+
+function methods:FadeInBackground(duration)
+	self.BackgroundTweener:Tween(duration, 0)
+	--self:FadeInText(duration)
+end
+
+function methods:FadeOutText(duration)
+	self.TextTweener:Tween(duration, 1)
+end
+
+function methods:FadeInText(duration)
+	self.TextTweener:Tween(duration, 0)
+end
+
+function methods:CreateTweeners()
+	self.BackgroundTweener:CancelTween()
+	self.TextTweener:CancelTween()
+
+	self.BackgroundTweener = moduleTransparencyTweener.new()
+	self.TextTweener = moduleTransparencyTweener.new()
+
+	--// Register BackgroundTweener objects and properties
+	self.BackgroundTweener:RegisterTweenObjectProperty(self.GuiObject, "BackgroundTransparency")
+	self.BackgroundTweener:RegisterTweenObjectProperty(self.TextBoxFrame, "BackgroundTransparency")
+
+	--// Register TextTweener objects and properties
+	local registerAsText = false
+	if (registerAsText) then
+		self.TextTweener:RegisterTweenObjectProperty(self.TextLabel, "TextTransparency")
+		self.TextTweener:RegisterTweenObjectProperty(self.TextLabel, "TextStrokeTransparency")
+		self.TextTweener:RegisterTweenObjectProperty(self.TextBox, "TextTransparency")
+		self.TextTweener:RegisterTweenObjectProperty(self.TextBox, "TextStrokeTransparency")
+	else
+		self.BackgroundTweener:RegisterTweenObjectProperty(self.TextLabel, "TextTransparency")
+		self.BackgroundTweener:RegisterTweenObjectProperty(self.TextLabel, "TextStrokeTransparency")
+		self.BackgroundTweener:RegisterTweenObjectProperty(self.TextBox, "TextTransparency")
+		self.BackgroundTweener:RegisterTweenObjectProperty(self.TextBox, "TextStrokeTransparency")
+	end
+
+end
+
+--///////////////////////// Constructors
+--//////////////////////////////////////
+ClassMaker.RegisterClassType("ChatBar", methods)
+
+function module.new()
+	local obj = {}
+	
+	local BaseFrame, TextBoxFrame, TextBox, TextLabel = CreateGuiObject()
+	obj.GuiObject = BaseFrame
+	obj.TextBoxFrame = TextBoxFrame
+	obj.TextBox = TextBox
+	obj.TextLabel = TextLabel
+
+	obj.TweenPixelsPerSecond = 500
+	obj.TargetYSize = 0
+
+	obj.BackgroundTweener = moduleTransparencyTweener.new()
+	obj.TextTweener = moduleTransparencyTweener.new()
+
+	ClassMaker.MakeClass("ChatBar", obj)
+
+	obj:SetTextLabelText("To chat click here or press \"/\" key")
```

Comment:
You can use single quotes or the [[ ]] string creation to avoid complicated escapes